common/static/common/templates/discussion/new-post.underscore

```diff
@@ -32,15 +32,15 @@
     <% } %>
     <div class="post-field group-visibility" id="wrapper-visibility-message"></div>
     <div class="post-field">
-        <label class="field-label">
+        <label class="field-label" for='discussion-forum-title'>
             <span class="field-label-text"><%- gettext("Title") %></span>
```

Comment:
As we are doing an explicit declaration of `label`, why we still need to include this `span` in this label here and in other file also?

Reply:
What about following a more granular approach by separating out label and other elements of html and only programmatically associate label to one the html element, say input in this case?

Reply:
for the span inside label, I was following the HTML for discussion and on some occasions, the span was used inside to display the actual text, but again there were places where text was just inside label tag. Since there were multiple methods by which label was showing actual text, I went with span inside label.

Reply:
For the second comment, do you mean that span and label shouldn't be nested? What I understand is that you suggest to move out span and keep it independent. Is that correct?

Reply:
Yes, if it is possible. I didn't get why we need to keep only span inside the label but took every other element out.

Reply:
Since we moved from the implicit association to explicit, that's why label wrapping was removed from everything(particularly input tag). As for span inside label, it can be removed i.e. text displayed directly. But as I mentioned previously that there were different ways text was displayed on the label, I opted for the nesting. But you are also right that span can be removed or moved out of nesting.

Reply:
You don't need to stop using the wrapping <label> pattern. Just add the for= attribute (and associated ID).  The wrapping label is in some cases (like with radio buttons) a usability feature in that it makes the click/touch target larger for the associate <input> . -- an accessibility benefit for people who have dexterity issues.